src/libraries/System.Private.CoreLib/src/System/Array.cs

```diff
@@ -916,29 +916,23 @@ public static int BinarySearch(Array array, int index, int length, object? value
                                 result = GenericBinarySearch<ushort>(array, adjustedIndex, length, value);
                                 break;
                             case CorElementType.ELEMENT_TYPE_I4:
-#if TARGET_32BIT
-                            case CorElementType.ELEMENT_TYPE_I:
-#endif
                                 result = GenericBinarySearch<int>(array, adjustedIndex, length, value);
                                 break;
                             case CorElementType.ELEMENT_TYPE_U4:
-#if TARGET_32BIT
-                            case CorElementType.ELEMENT_TYPE_U:
-#endif
                                 result = GenericBinarySearch<uint>(array, adjustedIndex, length, value);
                                 break;
                             case CorElementType.ELEMENT_TYPE_I8:
-#if TARGET_64BIT
-                            case CorElementType.ELEMENT_TYPE_I:
-#endif
                                 result = GenericBinarySearch<long>(array, adjustedIndex, length, value);
                                 break;
                             case CorElementType.ELEMENT_TYPE_U8:
-#if TARGET_64BIT
-                            case CorElementType.ELEMENT_TYPE_U:
-#endif
                                 result = GenericBinarySearch<ulong>(array, adjustedIndex, length, value);
                                 break;
+                            case CorElementType.ELEMENT_TYPE_I:
+                                result = GenericBinarySearch<nint>(array, adjustedIndex, length, value);
+                                break;
+                            case CorElementType.ELEMENT_TYPE_U:
+                                result = GenericBinarySearch<nuint>(array, adjustedIndex, length, value);
```

Comment:
```suggestion
                                result = (IntPtr.Size == 4) ? GenericBinarySearch<int>(array, adjustedIndex, length, value) : GenericBinarySearch<long>(array, adjustedIndex, length, value);
                                break;
                            case CorElementType.ELEMENT_TYPE_U:
                                result = (IntPtr.Size == 4) ? GenericBinarySearch<uint>(array, adjustedIndex, length, value) : GenericBinarySearch<ulong>(array, adjustedIndex, length, value);
```
Same pattern as below - avoids nint/nuint generic instantiations.

Reply:
Could you please update all `#if TARGET_32/64BIT` in Array.cs to this pattern? We can merge this PR after that.